script.js

```diff
@@ -1,4 +1,142 @@
+  //   (playerObject == REVERSED_PAPER && computerObject == SCISSORS)||
+  //   (playerObject == REVERSED_STONE && computerObject == PAPER)
+  // );
+  if (
+    (playerObject == SCISSORS && computerObject == PAPER) ||
+    (playerObject == PAPER && computerObject == STONE) ||
+    (playerObject == STONE && computerObject == SCISSORS) ||
+    (playerObject == REVERSED_SCISSORS && computerObject == STONE)||
+    (playerObject == REVERSED_PAPER && computerObject == SCISSORS)||
+    (playerObject == REVERSED_STONE && computerObject == PAPER)
+    ) {
+      return true
+    }
+    return false
+};
+/**
+ * Set a fn that will return an icon based on a  given object
+ * @param {string} object
+ */
+var getObjectIcon = function (object) {
+  if (object == SCISSORS || object == REVERSED_SCISSORS) return ' ✂️';
+  if (object == PAPER || object == REVERSED_PAPER) return ' 🗒';
+  if (object == STONE || object == REVERSED_STONE) return ' 🪨';
 };
+/**
+ * Return standard string representing player's and computer's objects
+ * @param {*} playerObject
+ * @param {*} computerObject
+ */
+var getDefaultObjectsMessage = function (playerObject, computerObject) {
+  var playerObjectIcon = getObjectIcon(playerObject);
+  var computerObjectIcon = getObjectIcon(computerObject);
+  return (
+    'The computer chose ' +
+    computerObject +
+    computerObjectIcon +
+    '<br><br>' +
+    'You chose ' +
+    playerObject +
+    playerObjectIcon
+  );
+};
+/**
+ * Check whether player draws with computer
+ * @param {string} playerObject
+ * @param {string} computerObject
+ */
+var doesPlayerDrawWComputer = function (playerObject, computerObject) {
+  return (
+    playerObject == computerObject ||
+    (playerObject == REVERSED_SCISSORS && computerObject == SCISSORS) ||
+    (playerObject == REVERSED_PAPER && computerObject == PAPER) ||
+    (playerObject == REVERSED_STONE && computerObject == STONE)
+  );
+};
+/**
+ * Play SPS with user input, return game result.
+ * @param {string} input - Player's object (e.g. scissors, paper, stone)
+ */
+var main = function (input) {
+  // Validate that input is one of scissors, paper, or stone
+  if (
+    input != SCISSORS &&
+    input != PAPER &&
+    input != STONE &&
+    input != REVERSED_SCISSORS &&
+    input != REVERSED_PAPER &&
+    input != REVERSED_STONE
+  ) {
+    return 'Please input 1 of "scissors", "paper", or "stone" to play the game.';
+    // Side enote: recall what "return" statements do? <--ans: it exits the current function and returns the values that are included in the return statement.
+    // This implies that if the return statement in line 107 is triggered, the rest of the code (from line 109 onwards) won't be executed becos we would've exited from the function.
+  }
+
+  // Input is one of scissors, paper, or stone
+  // Call input playerObject to make game logic clearer
+  var playerObject = input;
+  // Get computer's computer-generated object
+  var computerObject = getRandomObject();
```

Comment:
Good use of helper functions (in several lines below too) to abstract away the implementation details from main. This leaves your main looking concise, with just enough content for the reader to get a big picture view of what your code is doing.